runtime/js/40_command.js

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+"use strict";
+
+((window) => {
+  const core = window.Deno.core;
+  const { pathFromURL } = window.__bootstrap.util;
+  const { read, write } = window.__bootstrap.io;
+  const { illegalConstructorKey } = window.__bootstrap.webUtil;
+  const { ArrayPrototypeMap, ObjectEntries, String, TypeError } =
+    window.__bootstrap.primordials;
+
+  function createWritableIOStream(rid) {
+    return new WritableStream({
+      async write(chunk) {
+        let nwritten = 0;
+        while (nwritten < chunk.byteLength) {
+          nwritten += await write(rid, chunk.subarray(nwritten));
+        }
+      },
+      abort() {
+        core.tryClose(rid);
+      },
+    });
+  }
+
+  function createReadableIOStream(rid) {
+    return new ReadableStream({
+      async pull(controller) {
+        const view = controller.byobRequest.view;
+        const res = await read(rid, view);
+        if (res === null) {
+          core.close(rid);
+          controller.close();
+        } else {
+          controller.byobRequest.respond(res);
+        }
+      },
+      cancel() {
+        core.close(rid);
+      },
+      type: "bytes",
+      autoAllocateChunkSize: 16384,
+    });
+  }
+
+  class Command {
+    #options;
+
+    constructor(command, {
+      args = [],
+      cwd = undefined,
+      clearEnv = false,
+      env = {},
+      gid = undefined,
+      uid = undefined,
+    } = {}) {
+      this.#options = {
+        cmd: pathFromURL(command),
+        args: ArrayPrototypeMap(args, String),
+        cwd: pathFromURL(cwd),
+        clearEnv,
+        env: ObjectEntries(env),
+        gid,
+        uid,
+      };
+    }
+
+    spawn(options = {}) {
+      const child = core.opSync("op_command_spawn", {
+        ...this.#options,
+        ...options,
+      });
+
+      return new Child(illegalConstructorKey, child);
+    }
+
+    async status(options = {}) {
+      const status = await core.opAsync("op_command_status", {
+        ...this.#options,
+        ...options,
+      });
+      // TODO(@crowlKats): change typings to return null instead of undefined for status.signal
```

Comment:
I'm don't understand what you mean here

Reply:
`Deno.ProcessStatus` has a `signal` property that is defined as `signal?: undefined | number`, but since i create `Deno.ProcessStatus` in rust, it will type as `signal: null | number` instead (since `None` turns into `null`, not `undefined`)

Reply:
I believe the preferable solution would be to map `null`s to `undefined`

Reply:
so leave as-is and remove the 2.0 todo comments?

Reply:
I meant that if Rust returns `null` it should be mapped into `undefined` on the JS side. To it is `undefined | number` instead of `null | number`. Though I might be wrong. Would be great to have a second opinion on this one.